activation/atxdb.go

```diff
@@ -457,20 +426,32 @@ func (db *ActivationDb) GetEpochAtxIds(epochId types.EpochId) ([]types.AtxId, er
 // getAtxUnlocked gets the atx from db, this function is not thread safe and should be called under db lock
 // this function returns a pointer to an atx and an error if failed to retrieve it
 func (db *ActivationDb) getAtxUnlocked(id types.AtxId) (*types.ActivationTx, error) {
+	if atx, gotIt := db.atxCache.Get(id); gotIt {
+		return atx, nil
+	}
 	b, err := db.atxs.Get(id.Bytes())
 	if err != nil {
 		return nil, err
 	}
-	atx, err := types.BytesAsAtx(b)
+	atx, err := types.BytesAsAtx(b, &id)
 	if err != nil {
 		return nil, err
 	}
+	db.atxCache.Add(id, atx)
 	return atx, nil
 }
 
 // GetAtx returns the atx by the given id. this function is thread safe and will return error if the id is not found in the
 // atx db
-func (db *ActivationDb) GetAtx(id types.AtxId) (*types.ActivationTx, error) {
+func (db *ActivationDb) GetAtx(id types.AtxId) (*types.ActivationTxHeader, error) {
+	atx, err := db.GetFullAtx(id)
```

Comment:
do we really need to read the entire ATX? Does it also read the raw nipst?

Reply:
As discussed, we'll do this in a separate PR (issue created: https://github.com/spacemeshos/go-spacemesh/issues/1243)